tests/lib/rules/no-restricted-imports.js

```diff
@@ -1020,6 +1020,199 @@ ruleTester.run("no-restricted-imports", rule, {
+                line: 1,
+                column: 20,
+                endColumn: 23
+            }
+        ]
+    },
+    {
+        code: "import * as mod from 'mod'",
+        options: [{
+            paths: [
+                {
+                    name: "mod",
+                    importNames: ["foo"],
+                    message: "Import foo from qux instead."
+                },
+                {
+                    name: "mod",
+                    importNames: ["bar"],
+                    message: "Import bar from qux instead."
+                }
+            ]
+        }],
+        errors: [
+            {
+                message: "* import is invalid because 'foo' from 'mod' is restricted. Import foo from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 8,
+                endColumn: 16
+            },
+            {
+                message: "* import is invalid because 'bar' from 'mod' is restricted. Import bar from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 8,
+                endColumn: 16
+            }
+        ]
+    },
+    {
+        code: "import { foo } from 'mod'",
+        options: [{
+            paths: [
+
+                // restricts importing anything from the module
+                {
+                    name: "mod"
+                },
+
+                // message for a specific import name
+                {
+                    name: "mod",
+                    importNames: ["bar"],
+                    message: "Import bar from qux instead."
+                }
+            ]
+        }],
+        errors: [
+            {
+                message: "'mod' import is restricted from being used.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 1,
+                endColumn: 26
+            }
+        ]
+    },
+    {
+        code: "import { bar } from 'mod'",
+        options: [{
+            paths: [
+
+                // restricts importing anything from the module
+                {
+                    name: "mod"
+                },
+
+                // message for a specific import name
+                {
+                    name: "mod",
+                    importNames: ["bar"],
+                    message: "Import bar from qux instead."
+                }
+            ]
+        }],
+        errors: [
+            {
+                message: "'mod' import is restricted from being used.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 1,
+                endColumn: 26
+            },
+            {
+                message: "'bar' import from 'mod' is restricted. Import bar from qux instead.",
+                type: "ImportDeclaration",
+                line: 1,
+                column: 10,
+                endColumn: 13
+            }
+        ]
+    },
```

Comment:
This produces two errors: one for the whole import, and another specifically for the imported name `bar`. I think it's fine and technically correct.

Reply:
Agreed.